test/jdk/javax/swing/border/EtchedBorder/ScaledEtchedBorderTest.java

```diff
@@ -66,165 +71,256 @@ public class ScaledEtchedBorderTest {
+                        if (color == HIGHLIGHT.getRGB()) {
+                            borderThickness++;
+                        } else if (color == INSIDE_COLOR.getRGB()) {
+                            if (borderThickness != thickness) {
+                                throwWrongThickness(thickness, borderThickness, x, y);
+                            }
+                            borderThickness = 0;
+                            state = State.INSIDE;
+                        } else {
+                            throwUnexpectedColor(x, y, color);
+                        }
+                        break;
+
+                    case INSIDE:
+                        if (color == SHADOW.getRGB()) {
+                            state = State.RIGHT_SHADOW;
+                            borderThickness = 1;
+                        } else if (color != INSIDE_COLOR.getRGB()) {
+                            throwUnexpectedColor(x, y, color);
+                        }
+                        break;
+
+                    case RIGHT_SHADOW:
+                        if (color == SHADOW.getRGB()) {
+                            borderThickness++;
+                        } else if (color == HIGHLIGHT.getRGB()) {
+                            if (borderThickness != thickness) {
+                                throwWrongThickness(thickness, borderThickness, x, y);
+                            }
+                            borderThickness = 1;
+                            state = State.RIGHT_HIGHLIGHT;
+                        } else {
+                            throwUnexpectedColor(x, y, color);
+                        }
+                        break;
+
+                    case RIGHT_HIGHLIGHT:
+                        if (color == HIGHLIGHT.getRGB()) {
+                            borderThickness++;
+                        } else if (color == OUTER_COLOR.getRGB()) {
+                            if (borderThickness != thickness) {
+                                throwWrongThickness(thickness, borderThickness, x, y);
+                            }
+                            borderThickness = 0;
+                            state = State.BACKGROUND;
+                        } else {
+                            throwUnexpectedColor(x, y, color);
+                        }
+                        break;
                 }
-            }
-        }
+            } while (yStep > 0 && ((y += yStep) < height));
+        } while (xStep > 0 && ((x += xStep) < width));
+    }
+
+    private enum State {
+        BACKGROUND,
+        LEFT_SHADOW, LEFT_HIGHLIGHT,
+        INSIDE,
+        RIGHT_SHADOW, RIGHT_HIGHLIGHT
+    }
+
+    private static void throwWrongThickness(int thickness, int borderThickness,
+                                            int x, int y) {
+        throw new Error(
+                String.format("Wrong border thickness at %d, %d: %d vs %d",
+                              x, y, borderThickness, thickness));
```

Comment:
Since `throwWrongThickness()` is called commonly for different state values, including the state value in the error msg could be useful for debugging purpose.